### ISSUE TITLE
Hide "dev flags" `--show-ssa`, `--show-brillig`, and `--print-acir` from showing in CLI help

### ISSUE THREAD

**Issue by TomAFrench** [noir-lang/noir#2130]
### Problem

The `--show-ssa`, `--show-brillig`, and `--print-acir` are not hugely useful for most users of Nargo, mainly for us on the Noir team which need to use them for debugging the Noir compiler itself.

While they should exist, we don't need to document them for end users.

### Happy Case

These flags should be hidden from the CLI help output. One exception on this is that we should potentially allow printing ACIR on the `nargo info` command.

### Would you like to submit a PR for this Issue?

Yes

**Comment by Savio-Sou** [comment]
What's the harm to leaving them as is? Noise?

**Comment by TomAFrench** [comment]
Yeah, 90% of developers never need to use these flags. I'm also planning on restricting which commands they apply to (they currently apply to `prove` just because it implicitly compiles Noir code if we don't provide ACIR).

**Comment by Savio-Sou** [comment]
Makes sense.

I guess I don't have a strong opinion for this case specifically, but in general we might want to have one "source of truth" (whether it being `nargo help`, docs, or other places) where people that want to dig deep can count on.

**Comment by kobyhallx** [comment]
While certain things would be available (but hidden) we can state in documentation that they are there but we believe they're not useful for wider population and therefore we do not think about continuing support for them; they might be exposed in some other ways...

**Comment by Savio-Sou** [comment]
@kobyhallx certainly!

Do check the doc needs box on the PR if docs are where the info should live 🙌
